### DISCUSSION TITLE
renovate-on-prem should filter cron and event triggers by repositories key

### DISCUSSION THREAD

**Discussion by dnwe** [mend/renovate-ce-ee#274] [Q&A]
Similarly to https://github.com/mend/renovate-on-prem/issues/202, I was very surprised to find that renovate-on-prem doesn't respect `autodiscover: false` and an explicit `repositories` tag from the configuration file. 

It's particularly unexpected as RENOVATE_AUTODISCOVER_FILTER _is_ respected by the cron trigger, allowing you to limit the repositories it'll iterate over using one or more regex filters.

It feels like the expected behaviour would be for the server.js code to drop events for repositories that are not explicitly described in the supplied config.js if autodiscover is false.

**Comment by rarkins** [comment]
Which platform?

**Comment by dnwe** [comment]
github

**Comment by rarkins** [comment]
On GitHub, Renovate On-Prem is an app, and the app already allows you to "Select repositories" instead of "All repositories". How come you would prefer to use the `config.js` to narrow this list instead of GitHub's more convenient UI for selecting repositories?

**Reply by dnwe** [reply]
a UI is rarely convenient at scale

a) as mentioned on #202 the UI is limited to 100 repos and in any case nobody wants to manually click that many repositories

b) we do everything via configuration as code and automation, there’s no way to modify GitHub app configuration via the terraform provider or similar. 

The config.js method is very convenient — an array of repo names

**Comment by dnwe** [comment]
Aside: are there any plans to opensource the typescript behind the `server.js` code that ships as part of renovate-on-prem?

**Reply by rarkins** [reply]
No plans
